swupd: Require HTTPS/FILE protocol for mirror

jwakre opened the pull request:
Fixes Issue: #473

Changes proposed in this pull request:
- Return error when swupd-contenturl argument does not use HTTPS or FILE protocol
- Prevent gui/tui from accepting URL that is not the HTTPS or FILE protocol
- Change invalid URL string for gui/tui to suggest HTTPS: "Invalid URL" -> "Invalid URL: Use HTTPS"
- Update locales **(I am not confident in my Chinese translation, so I did not update this)**
- Mixed/Uppercase HTTP and FILE protocols are accepted (Swupd accepts Mixed/Uppercase)

jwakre commented on locale/zh_CN/LC_MESSAGES/clr-installer.po:
```diff
@@ -655,7 +655,7 @@ msgstr "启用自动更新"
 msgstr "警告: 禁用自动操作系统更新会使系统面临丢失关键安全修补程序的风险。"
 
-msgid "Invalid URL"
+msgid "Invalid URL: Use HTTPS"
```
My online translation was much longer than the existing translation, so I did not change it.

reaganlo replied:
Sorry I did not understand this. Why are we not having the full Chinese translation **无效 URL:使用 HTTPS**

jwakre replied:
I didn't trust my translation: 網址無效：使用HTTPS. I can copy the one from your post.

reaganlo commented on swupd/swupd.go:
```diff
@@ -66,7 +67,7 @@ const (
 
 	// InvalidURL specifies invalid url error message
-	InvalidURL = "Invalid URL"
+	InvalidURL = "Invalid URL: Use HTTPS"
```
@mdhorn Is it ok that we are not explicitly mentioning **FILE** as one of the allowed URI schemes for the swupd mirror?

reaganlo commented on swupd/swupd.go:
```diff
@@ -525,6 +526,22 @@ func UnSetHostMirror() (string, error) {
+
+	httpPrefix := strings.HasPrefix(strings.ToLower(mirror), "https:")
+	filePrefix := strings.HasPrefix(strings.ToLower(mirror), "file:")
+	if httpPrefix != true && filePrefix != true {
```
@mdhorn @otaviobp So we are going to limit the user to using only **HTTPS** and **FILE** URI schemes as the swupd mirror in the Installer. Is that OK?

jwakre commented on swupd/swupd.go:
```diff
@@ -525,6 +526,22 @@ func UnSetHostMirror() (string, error) {
+		return false
+	}
+
+	httpPrefix := strings.HasPrefix(strings.ToLower(mirror), "https:")
```
s/http/https

jwakre commented:
This PR has been updated with the following changes:
* Added Chinese translation
* Renamed httpPrefix to httpsPrefix

It was decided that the installer will suggest HTTPS, but it will also accept FILE. Other protocols will not be accepted.

jwakre commented:
There are concerns about swupd's restrictions for the HTTP protocol. I am going to hold-off on merging this PR until these concerns are resolved.

jwakre commented:
> There are concerns about swupd's restrictions for the HTTP protocol. I am going to hold-off on merging this PR until these concerns are resolved.

This PR is acceptable, but support for HTTP mirrors is still desired. I created #478 to add HTTP mirror support. Users will be required to acknowledge (a checkbox or flag) that they are enabling insecure downloads to enable HTTP.
